### PR TITLE
Fix `TaggedTemplateLiteral` evaluation

### PR DESCRIPTION
**Pull request by nmn**
This line of code causes a type error since `TaggedTemplateLiteral`s do not have a property called "expressions".

This change does what is probably expected. However, to maintain the existing behaviour, the line should instead be:

```js
const exprs: Array<undefined | NodePath<t.Node>> = path.isTemplateLiteral()
    ? path.get('expressions')
    : []
```

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/15224"></a>

### CODE REVIEW

File: packages/babel-traverse/src/path/evaluation.ts [hunk 1]
```diff
@@ -430,7 +430,11 @@ function evaluateQuasis(
   let str = "";
 
   let i = 0;
-  const exprs = path.get("expressions");
+  const exprs: Array<NodePath<t.Node>> = path.isTemplateLiteral()
+    ? path.get('expressions')
+    : (path as NodePath<t.TaggedTemplateExpression>)
+        .get('quasi')
+        .get('expressions');
```
Comment (liuxingbaoyu) [inline packages/babel-traverse/src/path/evaluation.ts:437]:
```suggestion
  const exprs = path.isTemplateLiteral()
    ? path.get("expressions")
    : path.get("quasi.expressions");
```
Will this work fine?

**Review by nicolo-ribaudo — COMMENTED** [review]
Can you add a test in `packages/babel-traverse/test/evaluation.js`, to show the bug that this PR fixes?

File: packages/babel-traverse/src/path/evaluation.ts (hunk 1 above)
Comment (nmn) [reply packages/babel-traverse/src/path/evaluation.ts:437]: This will work fine, but I’m not sure if the types will work correctly with it.

**Comment by nmn** [comment]
> Can you add a test in `packages/babel-traverse/test/evaluation.js`, to show the bug that this PR fixes?

The primary reason for this fix is that it’s a type error. However I’ll see if I can add/update a test.

**Review by nicolo-ribaudo — APPROVED** [review]

File: packages/babel-traverse/test/evaluation.js [hunk 2]
```diff
@@ -100,12 +100,27 @@ describe("evaluation", function () {
...
+          .get("body.2.declarations.0.init")
+          .evaluate().value,
+      ).toBe("value is 4");
+    });
+
+    it("shold evaluate String.raw tags", function () {
+      expect(
+        getPath("String.raw`a\\n${1}\\u`;").get("body.0.expression").evaluate()
+          .value,
+      ).toBe("a\\n1\\u");
```
Comment (nicolo-ribaudo) [inline packages/babel-traverse/test/evaluation.js:116]: This test was failing on main, returning `a\\n\\u`.

**Review by liuxingbaoyu — APPROVED** [review]

**Pull request merged**